### PR TITLE
Patch: Separation of secrets and cron

### PR DESCRIPTION
**Pull request by Psycho0verload**
#### Description
During the extensive changes with the S6 PullRequest #303, we have already discussed that the functionality of the Secrets should be retained, but that we would like to separate them cleanly in the end, when we have dealt with S6 a little more.

After @modem7 drew my attention to another project (https://github.com/linuxserver/docker-baseimage-alpine/blob/master/root/etc/s6-overlay/s6-rc.d/init-envfile/run) and I had a look at the changes made by @chrisblech - I decided to try to separate the cron and the secrets cleanly.

#### Tasks
- [x] Remove deprecated items
  - [x] Docker has [deprecated](https://github.com/compose-spec/compose-spec/blob/master/01-status.md#requirements-and-optional-attributes) the use of `version`
- [x] Separation of Cron and Secrets
  - [x] Own S6 service for Secrets 
  - [x] Optimizing debugging for secrets
  - [x] Make secrets available
  - [x] Error in cron script corrected (Thanks to @chrisblech)
- [x] Documentation adapted

#### Relevant resources
- https://github.com/linuxserver/docker-baseimage-alpine/blob/master/root/etc/s6-overlay/s6-rc.d/init-envfile/run
- https://github.com/caronc/apprise/wiki/Notify_email

#### Testing
There are several points to check with this PullRequest.
- [ ] Secrets work
- [ ] Secrets are displayed adequately with debugging activated
- [ ] Cron is set properly in the log
- [ ] Cron runs without errors 
- [ ] BACKUP_CRON behaves as actually defined

Here is an example docker-compose.yml template for testing:
```
services:
  borgmatic:
    image: borgmatic:s6-secrets-1000
    container_name: borgmatic
    volumes:
      - ${VOLUME_SOURCE}:/mnt/source:ro                     # backup source
      - ${VOLUME_TARGET}:/mnt/borg-repository            # backup target
      - ${VOLUME_ETC_BORGMATIC}:/etc/borgmatic.d/  # borgmatic config file(s) + crontab.txt
      - ${VOLUME_BORG_CONFIG}:/root/.config/borg      # config and keyfiles
      - ${VOLUME_SSH}:/root/.ssh                                     # ssh key for remote repositories
      - ${VOLUME_BORG_CACHE}:/root/.cache/borg       # checksums used for deduplication
    environment:
      - TZ=${TZ}
      - DEBUG_SECRETS=1
      - BACKUP_CRON=* * * * *
      - BORG_PASSPHRASE_FILE=/run/secrets/borg_passphrase
      - BORG_TEST_FILE=/run/secrets/borg_passphrase
      - YOUR_TEST_PASSHRASE_FILE=/run/secrets/borg_passphrase
    secrets:
      - borg_passphrase
secrets:
  borg_passphrase:
    file: ./secrets/secret
 ```

Feel free to test it extensively and ask questions.

### CODE REVIEW

**Comment by modem7** [comment]
FYI, I haven't forgotten this. I've been swamped with work. Will have a proper look when I can get 5 mins to myself!

**Review by grantbevis — APPROVED** [review]
LGTM 🚀 Has this been tested @Psycho0verload ?

Will await @modem7 's tick too

**Comment by Psycho0verload** [comment]
Works on my machine 😅
Tested it and add relevant documentations

File: docker-compose.restore.yml [hunk 1]
```diff
@@ -1,19 +1,13 @@
...
-
     security_opt:
       - apparmor:unconfined
       - label:disable
-
     devices:
       - /dev/fuse:/dev/fuse
-
-    command: /bin/sh
+    command: /bin/sh
```
Comment (modem7) [inline docker-compose.restore.yml:13]: We have bash available to us now just FYI. This may allow for more advanced commands over Ash.

**Review by modem7 — CHANGES_REQUESTED** [review]
I'd say the only thing to change is Ash to Bash unless there's a reason for using Ash. But otherwise? LGTM!

File: docker-compose.restore.yml (hunk 1 above)
Comment (Psycho0verload) [reply docker-compose.restore.yml:13]: @modem7 What else should I adjust in the file - I have actually only removed empty paragraphs 🧐😅

Comment (modem7) [reply docker-compose.restore.yml:13]: @Psycho0verload This is what happens when you review things just after you've gotten up, but before coffee. I'm a muppet. Let's leave as is, and we can improve it later down the line, it's beyond the scope of this PR.

**Review by modem7 — APPROVED** [review]
LGTM

File: docker-compose.restore.yml (hunk 1 above)
Comment (modem7) [reply docker-compose.restore.yml:13]: Test

**Review by modem7 — APPROVED** [review]

**Comment by grantbevis** [comment]
This change will be in the `:master` image being currently built

**Pull request merged**